common/src/thrift/common.thrift

```diff
@@ -44,6 +44,12 @@ struct NetAddress {
   3: i32 dataPort
 }
 
+// Information about the RPC.
+struct RpcOptions {
+  // key used to identify retried RPCs
+  1: optional string key
```

Comment:
I am fine with id or key ... but I would prefer to avoid redundancy (e.g. rpcId in a struct called rpcOptions).